packages/core/src/RenderingEngine/StackViewport.ts

```diff
@@ -2927,9 +2927,13 @@ class StackViewport extends Viewport implements IStackViewport, IImagesLoader {
       imageURI = imageId.substring(colonIndex + 1);
     }
 
-    const endsWith = referencedImageId?.endsWith(imageURI);
-    if (endsWith) {
-      return endsWith;
+    const referenceColonIndex = referencedImageId.indexOf(':');
```

Comment:
can you use this one packages/core/src/utilities/imageIdToURI.ts

Reply:
Sure can, updated this check as well as the check above it (and fixed some redundancy there)